### PR TITLE
Allow validators to use any valid combination of gold commitments as stake

### LINKED ISSUE #135: Validators SBAT use any valid combination of deposits as stake
### Expected Behavior

If a Validator has sufficient sum of balance in deposits with sufficient notice period, they should be able to register.

### Current Behavior

Validator must provide as stake a single Deposit, hence with a single notice period.

### PR DESCRIPTION
**Pull request by tkporter**
### Description

Allow validators to use any valid combination of gold commitments as stake.
Contract functions to register validators/validator groups now require an array of notice periods whose cumulative value for notice periods >= the minimum period is used to determine if the minimum value is satisfied.

When registering a validator or validator group with the cli, multiple `--noticePeriod` flags are used to specify multiple notice periods

### Tested

Ran tests, made sure CLI commands ran with multiple noticePeriods

### Other changes

n/a

### Related issues

- Fixes #135 

### Backwards compatibility

Requires new contract deployment & cli deployment

### CODE REVIEW

File: packages/protocol/contracts/governance/Validators.sol [hunk 1]
```diff
@@ -271,12 +271,12 @@ contract Validators is IValidators, Ownable, ReentrancyGuard, Initializable, Usi
       // secp256k1 public key + BLS public key + BLS proof of possession
       publicKeysData.length == (64 + 48 + 96)
     );
-    bytes memory proofOfPossessionBytes = publicKeysData.slice(64, 48 + 96);
```
Comment (tkporter) [inline packages/protocol/contracts/governance/Validators.sol:274]: Had to get rid of this variable to get rid of a stack too deep error

File: packages/cli/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
   "extends": "../typescript/tsconfig.library.json",
   "compilerOptions": {
     "rootDir": "src",
-    "outDir": "lib"
+    "outDir": "lib",
+    "esModuleInterop": true
```
Comment (tkporter) [inline packages/cli/tsconfig.json:6]: required to get the CLI working, see https://github.com/celo-org/celo-monorepo/pull/795

File: packages/protocol/contracts/governance/Validators.sol [hunk 3]
```diff
@@ -241,8 +241,8 @@ contract Validators is IValidators, Ownable, ReentrancyGuard, Initializable, Usi
    * @param identifier An identifier for this validator.
    * @param name A name for the validator.
    * @param url A URL for the validator.
-   * @param noticePeriod The notice period of the Locked Gold commitment that meets the
-   *   requirements for validator registration.
+   * @param noticePeriods The notice period of the Locked Gold commitments that
```
Comment (m-chrzan) [inline packages/protocol/contracts/governance/Validators.sol]: "period" still singular

File: packages/protocol/test/governance/validators.ts [hunk 4]
```diff
@@ -155,7 +155,7 @@ contract('Validators', (accounts: string[]) => {
 
   describe('#setMinElectableValidators', () => {
     const newMinElectableValidators = minElectableValidators.plus(1)
-    it('should set the minimum deposit', async () => {
+    it('should set the minimum commitment', async () => {
```
Comment (m-chrzan) [inline packages/protocol/test/governance/validators.ts]: Looks like this string was already wrong in previous commits, this should talk about minimum electable validators, right?

File: packages/protocol/test/governance/validators.ts [hunk 5]
```diff
@@ -193,7 +193,7 @@ contract('Validators', (accounts: string[]) => {
 
   describe('#setMaxElectableValidators', () => {
     const newMaxElectableValidators = maxElectableValidators.plus(1)
-    it('should set the minimum deposit', async () => {
+    it('should set the minimum commitment', async () => {
```
Comment (m-chrzan) [inline packages/protocol/test/governance/validators.ts]: Ditto.

File: packages/protocol/test/governance/validators.ts [hunk 6]
```diff
@@ -338,41 +338,111 @@ contract('Validators', (accounts: string[]) => {
...
 
-    describe('when the account is already a registered validator', () => {
-      beforeEach(async () => {
+    describe.only('when multiple commitment notice periods are provided', () => {
+      it('should accept a sufficient combination of commitments as stake', async () => {
+        // create registrationRequirement.value different locked commitments each
+        // with value 1 and unique noticePeriods greater than registrationRequirement.noticePeriod
+        const commitmentCount = registrationRequirement.value
+        const noticePeriods = []
+        for (let i = new BigNumber(1); i.lte(commitmentCount); i = i.plus(1)) {
```
Comment (m-chrzan) [inline packages/protocol/test/governance/validators.ts]: I would iterate with regular JS numbers here and elsewhere.

**Review by m-chrzan — COMMENTED** [review]
Nice! Just a few comments.

File: packages/protocol/test/governance/validators.ts [hunk 7]
```diff
@@ -817,34 +875,25 @@ contract('Validators', (accounts: string[]) => {
...
 
       it('should revert', async () => {
         await assertRevert(
-          validators.registerValidatorGroup(
-            identifier,
-            name,
-            url,
-            registrationRequirement.noticePeriod
-          )
+          await validators.registerValidatorGroup(identifier, name, url, [
```
Comment (m-chrzan) [inline packages/protocol/test/governance/validators.ts]: No `await` here. This should fix protocol tests.

**Review by m-chrzan — APPROVED** [review]
Looks good to me pending passing tests!

**Pull request merged**